tests/static_map/static_map_test.cu

```diff
@@ -489,8 +488,43 @@ TEMPLATE_TEST_CASE_SIG("Shared memory static map",
                                d_keys_exist.data().get(),
                                d_keys_and_values_correct.data().get());
 
-    REQUIRE(none_of(d_keys_exist.begin(), d_keys_exist.end(), [] __device__(const bool key_found)
-{ return key_found;
+    REQUIRE(none_of(d_keys_exist.begin(), d_keys_exist.end(), [] __device__(const bool key_found) {
+      return key_found;
     }));
   }
 }
+
+template <typename K, typename V, std::size_t N>
+__global__ void shared_memory_hash_table_kernel(bool* key_found)
+{
+  namespace cg   = cooperative_groups;
+  using map_type = typename cuco::static_map<K, V, cuda::thread_scope_block>::device_mutable_view;
+  using find_map_type = typename cuco::static_map<K, V, cuda::thread_scope_block>::device_view;
+  __shared__ typename map_type::slot_type slots[N];
+  auto map = map_type::make_from_uninitialized_slots(cg::this_thread_block(), &slots[0], N, -1, -1);
+
+  auto g            = cg::this_thread_block();
+  std::size_t index = threadIdx.x + blockIdx.x * blockDim.x;
+  int rank          = g.thread_rank();
+
+  // insert {thread_rank, thread_rank} for each thread in thread-block
+  map.insert(cuco::pair<int, int>(rank, rank));
+  g.sync();
+
+  auto find_map       = find_map_type(slots, N, -1, -1);
+  auto retrieved_pair = find_map.find(rank);
+  if (retrieved_pair != find_map.end() && retrieved_pair->second == rank) {
+    key_found[index] = true;
+  }
+}
+
+TEMPLATE_TEST_CASE("Shared memory slots.", "", int32_t)
+{
+  constexpr std::size_t N = 256;
+  thrust::device_vector<bool> key_found(N, false);
+  shared_memory_hash_table_kernel<TestType, TestType, N><<<8, 32, 32>>>(key_found.data().get());
```

Comment:
I assume the dynamic shared memory size was an error?
```suggestion
  shared_memory_hash_table_kernel<TestType, TestType, N><<<8, 32>>>(key_found.data().get());
```

Reply:
Totally my fault!